### PR TITLE
feat: os segment

### PR DESCRIPTION
**Pull request by nwykes**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

Add osicon segment for displaying an OS specific icon

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: segment.go [hunk 1]
```diff
@@ -56,6 +56,8 @@ const (
 	ShellInfo SegmentType = "shell"
 	//Node writes which node version is currently active
 	Node SegmentType = "node"
+	//OS write os specific icon
+	OsIcon SegmentType = "osicon"
```
Comment (JanDeDobbeleer) [inline segment.go]: Can we rename this to Os? As people will be able to set any string (which could also be plain `"windows"` for example), it feels more natural that way. I would leave `WindowsIcon` etc as is as the other segments use the same context and most likely people will be adding an icon.

File: segment_osicon_test.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOsIcon(t *testing.T) {
+	env := new(MockedEnvironment)
+	env.On("getRuntimeGOOS", nil).Return("windows")
+	osicon := &osicon{env: env}
```
Comment (JanDeDobbeleer) [inline segment_osicon_test.go]: You can also assign properties to validate the exact icon, so differ between set and not set.
See [here](https://github.com/JanDeDobbeleer/oh-my-posh3/blob/main/segment_exit_test.go#L30) for example.

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: segment.go (hunk 1 above)
Comment (nwykes) [reply segment.go]: I originally had it as "os", but had a naming collision with the os pkg.  I'll try changing the segment name to os and call the internal struct something else.

File: segment_osicon_test.go (hunk 2 above)
Comment (nwykes) [reply segment_osicon_test.go]: wil do, thanks.

File: segment.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply segment.go]: That's the best solution indeed 👍🏻

**Comment by JanDeDobbeleer** [comment]
@nwykes first off, thanks for the contributions and coffee. You made my day yesterday yesterday and today with that 🙏🏻

A final request, can you rebase to main and remove that merge commit? I have a personal dislike against those things 😅

**Comment by nwykes** [comment]
> @nwykes first off, thanks for the contributions and coffee. You made my day yesterday yesterday and today with that 🙏🏻
> 
> A final request, can you rebase to main and remove that merge commit? I have a personal dislike against those things 😅

least I can do, since I'm now using it in powershell in windows and zsh in Mac and linux everywhere, replacing my old powerlevel10k setup.  More coffees to come I hope.

Still working on the changes, but I'll get it rebased before submitting again.

**Pull request closed without merging**